cmd/talisman/talisman.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"flag"
+
+	"github.com/portworx/talisman/pkg/apis/portworx.com/v1alpha1"
+	"github.com/portworx/talisman/pkg/cluster/px"
+	"github.com/portworx/talisman/pkg/version"
+	"github.com/sirupsen/logrus"
+)
+
+type pxOperation string
+
+const (
+	pxOperationUpgrade pxOperation = "upgrade"
+)
+
+// command line arguments
+var (
+	newPXImage           string
+	op                   string
+	dockerRegistrySecret string
+)
+
+func main() {
+	logrus.Infof("Running talisman: %v", version.Version)
+	flag.Parse()
+
+	if len(op) == 0 {
+		logrus.Fatalf("error: no operation given for the PX cluster")
+	}
+
+	switch pxOperation(op) {
+	case pxOperationUpgrade:
+		if len(newPXImage) == 0 {
+			logrus.Fatalf("error: no PX image specified for %s operation", op)
+		}
+
+		inst, err := px.NewPXClusterProvider(dockerRegistrySecret)
+		if err != nil {
+			logrus.Fatalf("failed to instantiate PX cluster provider. err: %v", err)
+		}
+
+		newSpec := &v1alpha1.Cluster{
```

Comment:
This should have a comment to say what it is doing, because I can't tell from the code why this is necessary.